test/metabase/api/collection_test.clj

```diff
@@ -240,7 +272,43 @@
+      (testing "Excludes other user collections"
+        (let [admin-collection (collection/user->personal-collection (mt/user->id :crowberto))
+              lucky-collection (collection/user->personal-collection (mt/user->id :lucky))]
+          (mt/with-temp* [Collection [ac {:name "Admin Child" :location (collection/location-path admin-collection)}]
+                          Collection [lc {:name "Lucky Child" :location (collection/location-path lucky-collection)}]
+                          Collection [a {:name "A"}]
+                          Collection [b {:name     "B"
+                                         :location (collection/location-path a)}]
+                          Collection [c {:name "C"}]]
+            (let [ids                   (set (map :id [admin-collection lucky-collection ac lc a b c]))
+                  admin-response        (mt/user-http-request :crowberto :get 200
+                                                              "collection/tree")
+                  admin-response-ex     (mt/user-http-request :crowberto :get 200
+                                                              "collection/tree?exclude-other-user-collections=true")
+                  non-admin-response    (mt/user-http-request :lucky :get 200
+                                                              "collection/tree")
+                  non-admin-response-ex (mt/user-http-request :lucky :get 200
+                                                              "collection/tree?exclude-other-user-collections=true")]
+              ;; By default, our admin can see everything
+              (is (= [{:name "A", :children [{:name "B", :children []}]}
+                      {:name "C", :children []}
+                      {:name "Crowberto Corv's Personal Collection", :children [{:name "Admin Child", :children []}]}
+                      {:name "Lucky Pigeon's Personal Collection", :children [{:name "Lucky Child", :children []}]}]
+                     (collection-tree-view ids admin-response)))
+              ;; When excluding other user collections, the admin only sees their own collections and shared collections
+              (is (=
+                   [{:name "A", :children [{:name "B", :children []}]}
+                    {:name "C", :children []}
+                    {:name "Crowberto Corv's Personal Collection", :children [{:name "Admin Child", :children []}]}]
+                   (collection-tree-view ids admin-response-ex)))
+              ;; A non admin only sees their own collections and shared collections no matter what
+              (is (= [{:name "A", :children [{:name "B", :children []}]}
+                      {:name "C", :children []}
+                      {:name "Lucky Pigeon's Personal Collection", :children [{:name "Lucky Child", :children []}]}]
+                     (collection-tree-view ids non-admin-response)
+                     (collection-tree-view ids non-admin-response-ex))))))))
```

Comment:
Separate `(is ...)` clauses are easier to follow if one of the actual values is unexpected.